cli/azd/pkg/apphost/generate.go

```diff
@@ -989,24 +997,17 @@ func (b *infraGenerator) Compile() error {
 			return fmt.Errorf("configuring ingress for project %s: %w", resourceName, err)
 		}
 
+		dotnetCli := dotnet.NewDotNetCli(commandRunner)
+		port, err := dotnetCli.GetTargetPort(context.Background(), project.Path, "Release")
```

Comment:
We may over time need to come up with a way to delay running this code if we don't actually need the `TargetPort` (i.e. during `provision` when we only need it during `deploy`) but this is good to start with.

If we do end up needing to do something here we can make the TargetPort property be a zero argument `func`.